File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        int answer = 1;
        for(int i = 0; i < clothes.length; i++) {
            map.put(clothes[i][1],map.getOrDefault(clothes[i][1],0) +1);
        }
        int index = 0;
        int[] categories = new int[map.size()];
        for(String key : map.keySet()) {
            categories[index++] = map.get(key);
        }
        for(int i : categories) {
            System.out.print(i + " ");
        }
        for(int i = 0; i < categories.length; i++) {
            answer *= (categories[i] + 1);
        }
        
        
        answer -= 1;
        
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 0;
    const obj = {};
    for (const c of clothes) {
        const [_,type] = c;
        obj[type] = (obj[type] || 0) + 1;
    }
    for (const key in obj) {
        obj[key]++;
    }
    answer = Object.values(obj).reduce((acc,cur) => acc * cur, 1) - 1;

    return answer;
}